Rustbuild fails with ./x.py test src/libtest --stage 1

I suspect that the dependencies for libtest need to be updated, but I'm not sure; it's also possible this isn't intended to work, but a more user-friendly error would be nice.

Full output:
```
    Finished debug [unoptimized + debuginfo] target(s) in 0.0 secs
Synchronizing submodule url for 'src/compiler-rt'
Synchronizing submodule url for 'src/jemalloc'
Synchronizing submodule url for 'src/liblibc'
Synchronizing submodule url for 'src/llvm'
Synchronizing submodule url for 'src/rt/hoedown'
Synchronizing submodule url for 'src/rust-installer'
HEAD is now at 3bc0272 Merge pull request #26 from TimNN/arm-cc
HEAD is now at e058ca6 Change how the default zone is found
HEAD is now at 7d9b71f Merge pull request #435 from raphlinus/aarch64
HEAD is now at c1d9622 Backport r285278 [ARM] Predicate UMAAL selection on hasDSP.
HEAD is now at a3736a0 Merge pull request #6 from intelfx/patch-1
HEAD is now at 4f99485 Merge pull request #54 from brson/docdir
Building stage0 std artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.0 secs
Building stage0 test artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.0 secs
Building stage0 compiler artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.0 secs
Building stage1 std artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.0 secs
Building stage1 test artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.0 secs
Testing libtest stage1 (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
   Compiling test v0.0.0 (file:///home/mark/Edit/rust/src/libtest)
error[E0463]: can't find crate for `getopts`
  --> src/libtest/lib.rs:44:1
   |
44 | extern crate getopts;
   | ^^^^^^^^^^^^^^^^^^^^^ can't find crate

error: aborting due to previous error

error: Could not compile `test`.

To learn more, run the command again with --verbose.


command did not execute successfully: "/home/mark/Edit/rust/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "test" "-j" "8" "--target" "x86_64-unknown-linux-gnu" "--release" "--manifest-path" "/home/mark/Edit/rust/src/rustc/test_shim/Cargo.toml" "--features" "" "-p" "test"
expected success, got: exit code: 101
```

In fact, this sort of error seems to happen whenever I try to do tests after a particular stage:

```
$ ./x.py test src/test/compile-fail src/test/run-pass
    Finished debug [unoptimized + debuginfo] target(s) in 0.0 secs
Synchronizing submodule url for 'src/compiler-rt'
Synchronizing submodule url for 'src/jemalloc'
Synchronizing submodule url for 'src/liblibc'
Synchronizing submodule url for 'src/llvm'
Synchronizing submodule url for 'src/rt/hoedown'
Synchronizing submodule url for 'src/rust-installer'
HEAD is now at 3bc0272 Merge pull request #26 from TimNN/arm-cc
HEAD is now at e058ca6 Change how the default zone is found
HEAD is now at 7d9b71f Merge pull request #435 from raphlinus/aarch64
HEAD is now at c1d9622 Backport r285278 [ARM] Predicate UMAAL selection on hasDSP.
HEAD is now at a3736a0 Merge pull request #6 from intelfx/patch-1
HEAD is now at 4f99485 Merge pull request #54 from brson/docdir
Building stage0 std artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.0 secs
Building stage0 test artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.0 secs
Building stage0 compiler artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.0 secs
Building stage1 std artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.0 secs
Building stage1 test artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.0 secs
Building stage1 compiler artifacts (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
   Compiling rustc_platform_intrinsics v0.0.0 (file:///home/mark/Edit/rust/src/librustc_platform_intrinsics)
   Compiling serialize v0.0.0 (file:///home/mark/Edit/rust/src/libserialize)
   Compiling rustc_llvm v0.0.0 (file:///home/mark/Edit/rust/src/librustc_llvm)
   Compiling syntax_pos v0.0.0 (file:///home/mark/Edit/rust/src/libsyntax_pos)
   Compiling rustc_data_structures v0.0.0 (file:///home/mark/Edit/rust/src/librustc_data_structures)
   Compiling rustc_errors v0.0.0 (file:///home/mark/Edit/rust/src/librustc_errors)
error[E0463]: can't find crate for `term`
  --> src/librustc_errors/lib.rs:30:1
   |
30 | extern crate term;
   | ^^^^^^^^^^^^^^^^^^ can't find crate

error: aborting due to previous error

Build failed, waiting for other jobs to finish...
error: Could not compile `rustc_errors`.

To learn more, run the command again with --verbose.


command did not execute successfully: "/home/mark/Edit/rust/build/x86_64-unknown-linux-gnu/stage0/bin/cargo" "build" "-j" "8" "--target" "x86_64-unknown-linux-gnu" "--release" "--features" " jemalloc" "--manifest-path" "/home/mark/Edit/rust/src/rustc/Cargo.toml"
expected success, got: exit code: 101
```

I get the same error:

```
   Compiling rustc_errors v0.0.0 (file:///home/sanxiyn/rust/src/librustc_errors)
error[E0463]: can't find crate for `term`
```

--- a/src/bootstrap/util.rs
+++ b/src/bootstrap/util.rs
@@ -41,6 +41,12 @@ pub fn mtime(path: &Path) -> FileTime {
 /// Copies a file from `src` to `dst`, attempting to use hard links and then
 /// falling back to an actually filesystem copy if necessary.
 pub fn copy(src: &Path, dst: &Path) {
+    // A call to `hard_link` will fail if `dst` exists, so remove it if it
+    // already exists so we can try to help `hard_link` succeed.
+    let _ = fs::remove_file(&dst);
+
+    // Attempt to "easy copy" by creating a hard link (symlinks don't work on
+    // windows), but if that fails just fall back to a slow `copy` operation.
     let res = fs::hard_link(src, dst);
     let res = res.or_else(|_| fs::copy(src, dst).map(|_| ()));
     if let Err(e) = res {

